### ISSUE TITLE
app-i18n/mozc-2.28.5029.102 fails to build with dev-libs/protobuf-23.3-r2 and dev-cpp/abseil-cpp-20230802.0

### ISSUE THREAD

**Issue by mtaciano** [Gentoo-zh/gentoo-zh#3629]
`mozc-2.28.5029.102` is failing to build in my system, here's where it fails:

```
[381/534] /usr/bin/x86_64-pc-linux-gnu-g++ -pthread -o gen_usage_rewriter_dictionary_main -Wl,--start-group obj.host/rewriter/gen_usage_rewriter_dictionary_main.gen_usage_rewriter_dictionar
y_main.o obj.host/base/libbase.a obj.host/base/libbase_core.a obj.host/base/libserialized_string_array.a obj.host/base/libclock.a obj.host/base/libsingleton.a obj.host/base/libhash.a -Wl,--
end-group  -labsl_base -labsl_city -labsl_hash -labsl_malloc_internal -labsl_raw_hash_set -labsl_raw_logging_internal -labsl_spinlock_wait -labsl_status -labsl_strings -labsl_strings_intern
al -labsl_throw_delegate -labsl_debugging_internal -labsl_synchronization -labsl_time -labsl_civil_time -labsl_synchronization -labsl_time -labsl_time_zone -labsl_flags -labsl_flags_interna
l -labsl_synchronization -labsl_flags_reflection -labsl_flags_marshalling -labsl_str_format_internal -labsl_strings -labsl_strings_internal -labsl_flags_internal -labsl_status -labsl_flags_
commandlineflag_internal -labsl_flags_usage -labsl_flags_usage_internal -labsl_flags_internal -labsl_flags -labsl_flags_private_handle_accessor -labsl_flags_program_name -labsl_flags_parse
-labsl_random_distributions -labsl_random_internal_pool_urbg -labsl_random_internal_randen -labsl_random_internal_randen_hwaes -labsl_random_internal_randen_hwaes_impl -labsl_random_interna
l_randen_slow -labsl_random_internal_seed_material -labsl_random_seed_gen_exception -labsl_random_seed_sequences -labsl_status -labsl_statusor -labsl_synchronization -labsl_flags -labsl_gra
phcycles_internal
FAILED: gen_usage_rewriter_dictionary_main
```

```
/usr/bin/x86_64-pc-linux-gnu-g++ -pthread -o gen_usage_rewriter_dictionary_main -Wl,--start-group obj.host/rewriter/gen_usage_rewriter_dictionary_main.gen_usage_rewriter_dictionary_main.o o
bj.host/base/libbase.a obj.host/base/libbase_core.a obj.host/base/libserialized_string_array.a obj.host/base/libclock.a obj.host/base/libsingleton.a obj.host/base/libhash.a -Wl,--end-group
 -labsl_base -labsl_city -labsl_hash -labsl_malloc_internal -labsl_raw_hash_set -labsl_raw_logging_internal -labsl_spinlock_wait -labsl_status -labsl_strings -labsl_strings_internal -labsl_
throw_delegate -labsl_debugging_internal -labsl_synchronization -labsl_time -labsl_civil_time -labsl_synchronization -labsl_time -labsl_time_zone -labsl_flags -labsl_flags_internal -labsl_s
ynchronization -labsl_flags_reflection -labsl_flags_marshalling -labsl_str_format_internal -labsl_strings -labsl_strings_internal -labsl_flags_internal -labsl_status -labsl_flags_commandlin
eflag_internal -labsl_flags_usage -labsl_flags_usage_internal -labsl_flags_internal -labsl_flags -labsl_flags_private_handle_accessor -labsl_flags_program_name -labsl_flags_parse -labsl_ran
dom_distributions -labsl_random_internal_pool_urbg -labsl_random_internal_randen -labsl_random_internal_randen_hwaes -labsl_random_internal_randen_hwaes_impl -labsl_random_internal_randen_s
low -labsl_random_internal_seed_material -labsl_random_seed_gen_exception -labsl_random_seed_sequences -labsl_status -labsl_statusor -labsl_synchronization -labsl_flags -labsl_graphcycles_i
nternal
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: obj.host/base/base_core.file_util.o: undefined reference to symbol '_ZN4absl12lts_20230802lsERSoNS0_11string_view
E'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libabsl_string_view.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

```
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src/build_mozc.py", line 956, in <module>
    sys.exit(main())
             ^^^^^^
  File "/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src/build_mozc.py", line 941, in main
    BuildMain(cmd_opts, cmd_args)
  File "/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src/build_mozc.py", line 681, in BuildMain
    BuildWithNinja(options, targets)
  File "/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src/build_mozc.py", line 655, in BuildWithNinja
    RunOrDie([ninja, '-j12', '-l0', '-v', '-C', build_arg, target_name])
  File "/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src/build_tools/util.py", line 91, in RunOrDie
    raise RunOrDieError('\n'.join(['',
build_tools.util.RunOrDieError:
==========
 ERROR: ninja -j12 -l0 -v -C out_linux/Release mozc_server
==========
 * ERROR: app-i18n/mozc-2.28.5029.102::gentoo-zh failed (compile phase):
 *   Failed to run command: /usr/bin/python3.11 build_mozc.py build -c Release -v server/server.gyp:mozc_server unix/fcitx5/fcitx5.gyp:fcitx5-mozc gui/gui.gyp:mozc_tool
 *
 * Call stack:
 *     ebuild.sh, line  136:  Called src_compile
 *   environment, line 3023:  Called edo '/usr/bin/python3.11' 'build_mozc.py' 'build' '-c' 'Release' '-v' 'server/server.gyp:mozc_server' 'unix/fcitx5/fcitx5.gyp:fcitx5-mozc' 'gui/gui.gyp:
mozc_tool'
 *   environment, line 1128:  Called die
 * The specific snippet of code:
 *       "$@" || die -n "Failed to run command: $@"
 *
 * If you need support, post the output of `emerge --info '=app-i18n/mozc-2.28.5029.102::gentoo-zh'`,
 * the complete build log and the output of `emerge -pqv '=app-i18n/mozc-2.28.5029.102::gentoo-zh'`.
 * The complete build log is located at '/var/tmp/portage/app-i18n/mozc-2.28.5029.102/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/app-i18n/mozc-2.28.5029.102/temp/environment'.
 * Working directory: '/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src'
 * S: '/var/tmp/portage/app-i18n/mozc-2.28.5029.102/work/mozc-2.28.5029.102/src'
```

The main culprit seems to be the fact that the linker can't find the symbol `_ZN4absl12lts_20230802lsERSoNS0_11string_viewE`, which is obviously related to some string view function inside abseil version `20230802`.

First I thought that it was related to me using `mold` as a linker, so I disabled it for `mozc` and `abseil-cpp` and tried to recompile both, but nothing changed.

Then I tried to downgrade both `abseil-cpp` to `20230125.3-r1` and `protobuf` to `21.12` and the error changed, but I still couldn't compile.

There's some incompatibility with this `mozc` and `abseil-cpp` version. Also, it seems that there's a new version `2.29.5200.102` available for `mozc`.

**Comment by liangyongxiang** [comment]
https://bugs.gentoo.org/912776

**Comment by Universebenzene** [comment]
New version of protobuf breaks several packages. Seems no solution yet, except for downgrading protobuf.

**Comment by mtaciano** [comment]
After downgrading `dev-cpp/abseil-cpp` to `20230125.3-r1` and `dev-libs/protobuf` to both `21.12` and `21.9` I still couldn't compile. I'm currently encountering the following bug:  https://bugs.gentoo.org/913243

**Comment by Universebenzene** [comment]
@mtaciano How about protobuf-21.9 ?

**Comment by Universebenzene** [comment]
Should be fixed after #3661 merged. You can close this issue after confirming this.

**Comment by mtaciano** [comment]
Closing this issue as it now compiles without errors, fixed by #3661.
